### PR TITLE
Ensure inputValue is always up-to-date

### PR DESCRIPTION
**Pull request by edgifyjp**
This is a speculative PR for an issue I have found.

I was seeing an issue where the inputValue wasn't updating after TAB was used to autocomplete an option. Therefore valueChange() was reporting the wrong value.

I _think_ this PR fixes this issues, and I have a test to show it fixed, but I'm not sure that I understand the tokenising system as much as I should.

@bekzod Perhaps you could take a look and let me know if you think this is a safe fix, please?

Many thanks.

### CODE REVIEW

**Comment by bekzod** [comment]
looks great, I will have a look in detail later today thanks a lot

**Comment by edgifyjp** [comment]
Great. Thanks.

**Comment by bekzod** [comment]
@edgifyjp there is one thing broken though, in master if you check you can increment/decrement year/month/day when you have cursor on them http://cl.ly/1J1M3F3w0q3C/Screen%20Recording%202016-03-18%20at%2003.54%20PM.gif and in your branch it is not working (it looses focus)

File: addon/components/slack-search-input/component.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ export default Ember.Component.extend({
   inputValue: computed({
     set(key, val) {
       set(this, 'proxyValue', val);
+      set(this, 'cursorLocation', val.length);
```
Comment (bekzod) [inline addon/components/slack-search-input/component.js:37]: this might be causing the problem commented above

**Comment by edgifyjp** [comment]
@bekzod Thanks for spotting. I'll take a look.

**Comment by bekzod** [comment]
@edgifyjp merging this will will remove commented line though

**Comment by edgifyjp** [comment]
@bekzod Thanks for closing. I got the wrong end of the stick here. I'll take a better look at it.

**Pull request merged**